### PR TITLE
Alter a few tests to fix them within google3

### PR DESCRIPTION
**Pull request by rsgowman**
Without this change, these tests cause a:
"junit.framework.AssertionFailedError: android.content.res.Resources$NotFoundException: Unable to find resource ID #0x7f030000 "
error.

### CODE REVIEW

File: firebase-firestore/src/androidTest/java/com/google/firebase/firestore/FirestoreTest.java [hunk 1]
```diff
@@ -969,11 +967,9 @@ public void testMaintainsPersistenceAfterRestarting() {
     DocumentReference docRef = testDocument();
     FirebaseFirestore firestore = docRef.getFirestore();
     waitFor(docRef.set(map("foo", "bar")));
-    Context context = InstrumentationRegistry.getContext();
 
     waitFor(AccessHelper.shutdown(firestore));
-    FirebaseApp app2 = FirebaseApp.initializeApp(context);
-    FirebaseFirestore firestore2 = FirebaseFirestore.getInstance(app2);
+    FirebaseFirestore firestore2 = testFirestore();
```
Comment (wilhuff) [inline firebase-firestore/src/androidTest/java/com/google/firebase/firestore/FirestoreTest.java]: This change breaks the intent of the test.

`testFirestore()` creates a new Firestore instance associated with a new Firestore app. This guarantees that we'll use a different persistence key that the one we actually use. This test will pass even if clearPersistence is unimplemented.

The purpose of what we're doing here is attempting to create a new Firestore instance associated with the same app as the one before.

Comment (rsgowman) [reply firebase-firestore/src/androidTest/java/com/google/firebase/firestore/FirestoreTest.java]: > This test will pass even if clearPersistence is unimplemented.

Hmm... you're right. However, the *original* tests will also pass with an effectively empty clearPersistence. i.e.:
```
$ git diff
diff --git a/firebase-firestore/src/main/java/com/google/firebase/firestore/FirebaseFirestore.java b/firebase-firestore/src/main/java/com/google/firebase/firestore/FirebaseFirestore.java
index e935416..5021e17 100644
--- a/firebase-firestore/src/main/java/com/google/firebase/firestore/FirebaseFirestore.java
+++ b/firebase-firestore/src/main/java/com/google/firebase/firestore/FirebaseFirestore.java
@@ -404,17 +404,8 @@ public class FirebaseFirestore {
     if (this.clientRunning) {
       throw new IllegalStateException("Persistence cannot be cleared while the client is running.");
     }
-    final TaskCompletionSource<Void> source = new TaskCompletionSource<>();
-    asyncQueue.enqueueAndForget(
-        () -> {
-          try {
-            SQLitePersistence.clearPersistence(context, databaseId, persistenceKey);
-            source.setResult(null);
-          } catch (FirebaseFirestoreException e) {
-            source.setException(e);
-          }
-        });
-    return source.getTask();
+
+    return Tasks.forResult(null);
   }
 
   FirestoreClient getClient() {
$ ANDROID_HOME=~/Android/Sdk ./gradlew :firebase-firestore:connectedCheck
<successful output here>
```

I'll look into this a bit more tomorrow.

Comment (rsgowman) [reply firebase-firestore/src/androidTest/java/com/google/firebase/firestore/FirestoreTest.java]: Ok; I've reworked the tests. They now:
a) Properly fail when I noop clearPersistence (i.e. when I apply the above patch)
b) Pass within g3.

**Comment by rsgowman** [comment]
Where you referring to the error within g3? If so, unfortunately I have *no* idea. :)

Something in the depths of gmscore is requesting an android resource when initializing firebaseapp for the second time, and for reasons that are unclear to me, failing to find it. Digging into it further looked like a pretty large time sink, so I skipped it.

**Pull request merged**